contracts/cryptoblades.sol

```diff
@@ -271,6 +271,49 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         require(foundMatch, "Target invalid");
     }
 
+    mapping (address => uint32) tickets;
+
+    function spendTicket(uint32 num)
+        public
+    {
+        require (tickets[msg.sender] >= num, "You don't have enough tickets");
+
+        for (int i = 0; i < num; i++) {
+            weapons.mint(msg.sender, randoms.getRandomSeed(msg.sender));
```

Comment:
This will give every single weapon the exact same seed and thus make them identical. The random seed should be combined with `i` and passed to `keccak256()` to make a unique seed for each weapon.

Reply:
Fixed this.